backend/test/data_access_objects/test_app_configuration.py

```diff
@@ -0,0 +1,197 @@
+class TestAppConfigDAO(TestCase):
+    def setUp(self):
+        """
+        Set up virtual DDB resources/tables
+        """
+        from ml_space_lambda.data_access_objects.app_configuration import AppConfigurationModel
+        from ml_space_lambda.utils.common_functions import retry_config
+        from ml_space_lambda.utils.mlspace_config import get_environment_variables
+
+        env_vars = get_environment_variables()
+        self.TEST_TABLE = env_vars["APP_CONFIGURATION_TABLE"]
+        self.ddb = boto3.client(
+            "dynamodb",
+            config=retry_config,
+        )
+        self.ddb.create_table(
+            TableName=self.TEST_TABLE,
+            KeySchema=TEST_APP_CONFIG_TABLE_KEY_SCHEMA,
+            AttributeDefinitions=TEST_APP_CONFIG_TABLE_ATTRIBUTE_DEFINITIONS,
+            BillingMode="PAY_PER_REQUEST",
+        )
+        self.app_config_dao = AppConfigurationDAO(self.TEST_TABLE, self.ddb)
+
+        # Seed a global record and 2 project records
+        self.GLOBAL_RECORD = AppConfigurationModel.from_dict(generate_test_config("global", 0, False))
+        self.PROJECT_RECORD1 = AppConfigurationModel.from_dict(generate_test_config("project1", 0, True))
+        self.PROJECT_RECORD2 = AppConfigurationModel.from_dict(generate_test_config("project1", 1, True))
+
+        self.ddb.put_item(
+            TableName=self.TEST_TABLE,
+            Item=json.loads(dynamodb_json.dumps(self.GLOBAL_RECORD.to_dict())),
+        )
+
+        self.ddb.put_item(
+            TableName=self.TEST_TABLE,
+            Item=json.loads(dynamodb_json.dumps(self.PROJECT_RECORD1.to_dict())),
+        )
+
+        self.ddb.put_item(
+            TableName=self.TEST_TABLE,
+            Item=json.loads(dynamodb_json.dumps(self.PROJECT_RECORD2.to_dict())),
+        )
+
+    def tearDown(self):
+        """
+        Delete virtual DDB resources/tables
+        """
+        self.ddb.delete_table(TableName=self.TEST_TABLE)
+        self.ddb = None
+        self.app_config_dao = None
+
+    def test_get_app_config_global(self):
+        from_ddb = self.app_config_dao.get(self.GLOBAL_RECORD.configScope, 2)
+        # num_versions is set to 2, but there's only 1 record so we expect a list of len 1
+        assert len(from_ddb) == 1
+        assert from_ddb[0].to_dict() == self.GLOBAL_RECORD.to_dict()
+
+    def test_get_app_config_project(self):
+        from_ddb = self.app_config_dao.get(self.PROJECT_RECORD1.configScope, 2)
+        assert len(from_ddb) == 2
+        # The project configs should be returned in descending order based on version ID
+        # PROJECT_RECORD2 has a versionId of 2, so it should come first
+        assert from_ddb[0].to_dict() == self.PROJECT_RECORD2.to_dict()
+        assert from_ddb[1].to_dict() == self.PROJECT_RECORD1.to_dict()
+
+    def test_get_app_config_not_found(self):
+        assert not self.app_config_dao.get("sample-project", 1)
+
+    def test_create_app_config(self):
```

Comment:
We're not testing the case where we get a condition check failure. I know we've got a unit test for that but we should leverage moto to test that too since it's more representative then us mocking.

Reply:
I see what you mean, I'll add a test where we try to update using an existing versionId, we should expect a ConditionalCheckFailedException